packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -71,7 +67,7 @@ const getDeepPopulate = (uid, populate, depth = 0) => {
     if (attribute.type === 'dynamiczone') {
       populateAcc[attributeName] = {
         populate: (attribute.components || []).reduce((acc, componentUID) => {
-          return Object.assign(acc, getDeepPopulate(componentUID, null, depth + 1));
+          return merge(acc, getDeepPopulate(componentUID, null));
```

Comment:
I'm not sure I understand the use of merge ?

Reply:
A user wrote it: https://github.com/strapi/strapi/issues/11319#issuecomment-1026717218
I didn't know at the beginning but it's in case 2 components have a field with the same name. I thought it would not hurt to merge both instead of one overwriting the other (but I didn't try to be honest)

Reply:
@lukasz-mical would you have more insight?

Reply:
It looks like it always sub components of dz with the same names. Can you try that out to make sure it doesn't break something ?

Reply:
Nice